crates/wascc-provider/src/lib.rs

```diff
@@ -288,80 +292,75 @@ impl WasccProvider {
             )
         })
         .await?;
-        match http_result {
-            Ok(handle) => {
-                run_context
-                    .container_handles
-                    .insert(ContainerKey::App(container.name().to_string()), handle);
-                status_sender
-                    .broadcast(ContainerStatus::Running {
-                        timestamp: chrono::Utc::now(),
-                    })
-                    .expect("status should be able to send");
-                Ok(())
-            }
-            Err(e) => {
-                // We can't broadcast here because the receiver has been dropped at this point
-                // (it was never used in creating a runtime handle)
-                let mut container_statuses = HashMap::new();
-                container_statuses.insert(
-                    ContainerKey::App(container.name().to_string()),
-                    ContainerStatus::Terminated {
-                        timestamp: chrono::Utc::now(),
-                        failed: true,
-                        message: format!("Error while starting container: {:?}", e),
-                    },
-                );
-                let status = PodStatus {
-                    message: PodStatusMessage::LeaveUnchanged,
-                    container_statuses,
-                };
-                pod.patch_status(run_context.client.clone(), status).await;
-                Err(anyhow::anyhow!("Failed to run pod: {}", e))
-            }
-        }
+        http_result
     }
 }
 
-struct ModuleRunContext<'a> {
-    client: &'a kube::Client,
-    modules: &'a mut HashMap<String, Vec<u8>>,
-    volumes: &'a HashMap<String, Ref>,
-    container_handles: &'a mut ContainerHandleMap<ActorHandle, LogHandleFactory>,
+struct ModuleRunContext {
+    modules: HashMap<String, Vec<u8>>,
+    volumes: HashMap<String, Ref>,
+    event_tx: tokio::sync::mpsc::Sender<WatchEvent<KubePod>>,
+    event_rx: tokio::sync::mpsc::Receiver<WatchEvent<KubePod>>,
 }
 
-#[async_trait]
-impl Provider for WasccProvider {
-    const ARCH: &'static str = TARGET_WASM32_WASCC;
+#[async_trait::async_trait]
+impl DefaultStateProvider for WasccProvider {
```

Comment:
After reading through it here, I am not sure I really like this trait. Might be better to have things defined by the macros instead. Let's talk some more when we meet